core/src/main/scala/caliban/execution/Field.scala

```diff
@@ -217,31 +214,34 @@ object Field {
 
   private def resolveDirectiveVariables(
     variableValues: Map[String, InputValue],
-    variableDefinitions: List[VariableDefinition]
+    variableDefinitions: Map[String, VariableDefinition]
   )(directive: Directive): Directive =
     if (directive.arguments.isEmpty) directive
     else directive.copy(arguments = resolveVariables(directive.arguments, variableDefinitions, variableValues))
 
+  private val emptyVariables = Map.empty[String, InputValue]
```

Comment:
`Map.empty` already points to an object so you don't need to save it into a val.